backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendVmBackupResource.java

```diff
@@ -39,7 +39,12 @@ public BackendVmBackupsResource getParent() {
 
     @Override
     public Backup get() {
-        return addLinks(performGet(QueryType.GetVmBackupById, new IdQueryParameters(guid)));
+        Backup backup = performGet(QueryType.GetVmBackupById, new IdQueryParameters(guid));
+        if (backup.isSetSnapshot()) {
+            backup.getSnapshot().setHref(String.format("/ovirt-engine/api/vms/%s/snapshots/%s", vmId, backup.getSnapshot().getId()));
```

Comment:
@oliel I'm not sure about this, I saw this done in some places but it doesn't feel like the right way

Reply:
I see that LinkHelper::addLinks is more commonly used but Ori would know better :)

Reply:
yes, I tried it, it has something to do with the parent as it works for the VM, but not for the snapshot

https://github.com/oVirt/ovirt-engine/blob/3fe8f849c24485baa69d5a6704371d8ccf882aaa/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendVmBackupResource.java#L33

Reply:
I see

Reply:
LinkHelper checks which of the possible parents have an ID set within the entity. That's how it identifies the parent. When there is more than one reference with an ID, it goes arbitrarily with whichever it found first. This situation should be avoided by explicitly specifying the parent-type. If in this case the Snaphshot should be the parent, try: 

return addLinks(performGet(QueryType.GetVmBackupById, new IdQueryParameters(guid)), Snapshot.class);

Reply:
In this case Snapshot is not a parent of backup in the sense of there is a link /api/snapshots/<id>/backups (VM however is the parent of backup), rather a backup has a reference to snapshot (snapshot id) and I want to have a link to it like in the example I added above

Reply:
try this:
        Backup backup = performGet(QueryType.GetVmBackupById, new IdQueryParameters(guid));
        backup.getSnapshot().getVm().setId(vmId);
        addLinks(backup)

If it doesn't work try this:

        Backup backup = addLinks(performGet(QueryType.GetVmBackupById, new IdQueryParameters(guid)));
        backup.getSnapshot().getVm().setId(vmId);
        addLinks(backup)

Reply:
does the trick, thanks!